x-pack/legacy/plugins/lens/public/app_plugin/url_helper.ts

```diff
@@ -32,6 +30,42 @@ export function getKibanaBasePathFromDashboardUrl(url: string | undefined): stri
   return null;
 }
 
+/**
+ * Returns dashboard url with given query params. If query params already exist in the url, they will be replaced
+ * eg.
+ * input: http://localhost:5601/lib/app/kibana#/dashboard, {_a: {...}, _g: {...}}
+ * output: http://localhost:5601/lib/app/kibana#/dashboard??_g=(refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(description:'',filters:!(),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),panels:!(),query:(language:kuery,query:''),timeRestore:!f,title:'',viewMode:edit)
+ * @param absoluteUrl dashboard absolute url
+ * @param urlParams query params to append to the url
+ */
+export function getDashboardUrlWithQueryParams(
+  absoluteUrl: string,
+  urlParams: Record<string, string>
+): string {
+  let dashboardUrl = getUrlWithoutQueryParams(absoluteUrl);
+  if (!dashboardUrl) {
+    return absoluteUrl;
+  }
+  dashboardUrl += '?';
+  const keys = Object.keys(urlParams).sort();
+  keys.forEach((key, index) => {
+    dashboardUrl += `${key}=${urlParams[key]}`;
+    if (index !== keys.length - 1) {
+      dashboardUrl += '&';
+    }
+  });
+  return dashboardUrl;
+}
+
+export function getUrlVars(url: string): Record<string, string> {
+  const vars: Record<string, string> = {};
+  // @ts-ignore
+  url.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(_, key, value) {
+    vars[key] = value;
+  });
+  return vars;
+}
```

Comment:
When I was looking at use of rison in our code, I also found use of the `url` library and some other Kibana-specific URL parsing. Both this function and the previous function seem to be common functions in our code, can we simplify them?

Reply:
There is the old `kibanaUrl` thing, but that doesn't work in new platform afaik. Standard URL api would be the best, but that doesn't work in IE11 :( 
Anything specific you had in mind?

Reply:
I was hoping to find a way to get rid of the regexes and manual construction of URLs in favor of a standardized way. I'm not very familiar with the setup here, but it is still my assumption that there is a cleaner way of doing. I don't want to block this work over this concern though.